src/schemas/jsonSchema/index.js

```diff
@@ -0,0 +1,395 @@
+
+    return this.loadSchemaFromTitle(title[0]);
+  }
+
+  /**
+   * Adds a schema definition to be backwards compatible with the old schema structure.
+   */
+  async addDefinition(schema) {
+    if (/^credential-/.test(schema.title)) {
+      await this.addCredentialDefinition(schema);
+    } else {
+      await this.addClaimDefinition(schema);
+    }
+  }
+
+  /**
+   *
+   * @param schema
+   * @returns {Promise<void>}
+   */
+  async addCredentialDefinition(schema) {
+    const definition = {
+      identifier: schema.title,
+      version: getSchemaVersion(schema.title),
+      depends: [],
+    };
+
+    if (schema.transient) {
+      definition.transient = true;
+    }
+
+    if (schema.properties.claim.required) {
+      definition.required = [];
+    }
+
+    // TODO: clean this section
+    // eslint-disable-next-line guard-for-in,no-restricted-syntax
+    for (const k in schema.properties.claim.properties) {
+      const v = schema.properties.claim.properties[k];
+      // eslint-disable-next-line guard-for-in,no-restricted-syntax
+      for (const ki in v.properties) {
+        const vi = v.properties[ki];
+
+        // eslint-disable-next-line no-await-in-loop
+        const propertySchema = await this.loadSchemaFromUri(vi.$ref);
+
+        if (propertySchema !== null) {
+          definition.depends.push(propertySchema.title);
+        }
+
+        if (schema.properties.claim.required && schema.properties.claim.required.includes(ki)) {
+          definition.required.push(propertySchema.title);
+        }
+      }
+    }
+
+    this.credentialDefinitions.push(definition);
+    this.validCredentialIdentifiers.push(definition.identifier);
+    SummaryMapper.addCredentialDefinition(definition);
+  }
+
+  async addClaimDefinition(schema) {
+    const definition = {
+      identifier: schema.title,
+      version: getSchemaVersion(schema.title),
+      type: await this.findDefinitionType(schema),
+    };
+
+    if (definition.type === 'Array') {
+      const subSchema = await this.loadSchemaFromUri(schema.items.$ref);
+
+      definition.items = {
+        type: subSchema.title,
+      };
+    }
+
+    ['attestable', 'credentialItem', 'minimum', 'maximum'].forEach((property) => {
+      if (property in schema) {
+        definition[property] = schema[property];
+      }
+    });
+
+    if (schema.required) {
+      definition.type.required = schema.required;
+    }
+
+    this.definitions.push(definition);
+    this.validIdentifiers.push(schema.title);
+    SummaryMapper.addDefinition(definition);
+  }
+
+  async getPropertyValues(properties) {
+    const defProperties = [];
+
+    // eslint-disable-next-line guard-for-in,no-restricted-syntax
+    for (const name in properties) {
+      const property = properties[name];
+
+      let { type } = property;
+      const { items } = property;
+
+      if (type === 'array') {
+        if (items.$ref) {
+          // TODO: remove hardcode identifier
```

Comment:
What the plan for this TODO? There is already a story for the in the backlog? should we add one?

Reply:
Yeah, I've got a list of tickets to create from this (I'll add it)